src/shared/auth_client.c

```diff
@@ -0,0 +1,75 @@
+/*
+ * Local Authd client
+ * Copyright (C) 2017 Wazuh Inc.
+ * May 30, 2017.
+ *
+ * This program is a free software; you can redistribute it
+ * and/or modify it under the terms of the GNU General Public
+ * License (version 2) as published by the FSF - Free Software
+ * Foundation.
+ */
+
+#include "shared.h"
+#include <os_net/os_net.h>
+#include <external/cJSON/cJSON.h>
+
+// Remove agent. Returns 0 on success or -1 on error.
+int auth_remove_agent(int sock, const char *id, int json_format) {
+    char buffer[OS_MAXSTR + 1];
+    char *output;
+    int result;
+    ssize_t length;
+    cJSON *response;
+    cJSON *error;
+    cJSON *message;
+    cJSON *request = cJSON_CreateObject();
+    cJSON *arguments = cJSON_CreateObject();
+
+    cJSON_AddItemToObject(request, "arguments", arguments);
+    cJSON_AddStringToObject(request, "function", "remove");
+    cJSON_AddStringToObject(arguments, "id", id);
+
+    output = cJSON_PrintUnformatted(request);
+
+    if (OS_SendSecureTCP(sock, strlen(output), output) < 0) {
+        merror_exit("OS_SendSecureTCP(): %s", strerror(errno));
```

Comment:
Those `merror_exit()` functions would close Modulesd now. They should be an error or a warning, but never a critical error.